File: shopping-backend/src/main/java/shopping/entity/Product.java
package shopping.entity;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "products")
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "product_id")
    private Long productId;

    @Column(name = "description", nullable = false)
    private String description;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "quantity", nullable = false)
    private Integer quantity;

    @Column(name = "retail_price", nullable = false)
    private Double retailPrice;

    @Column(name = "wholesale_price", nullable = false)
    private Double wholesalePrice;

    @OneToMany(mappedBy = "product", fetch = FetchType.LAZY)
    private List<OrderItem> orderItems = new ArrayList<>();

    @OneToMany(mappedBy = "product", fetch = FetchType.LAZY)
    private List<Watchlist> watchlistItems = new ArrayList<>();
}

File: shopping-frontend/src/app/shared/models/product.ts
export interface Product {
    productId: number;
    description: string;
    name: string;
    retailPrice: number;
}

export interface AdminProduct {
    productId: number;
    description: string;
    productName: string;
    quantity: number;
    retailPrice: number;
    wholesalePrice: number;
}

export interface CreateProductRequest {
    description: string;
    name: string;
    quantity: number;
    retailPrice: number;
    wholesalePrice: number;
}

export interface UpdateProductRequest {
    description?: string;
    name?: string;
    quantity?: number;
    retailPrice?: number;
    wholesalePrice?: number;
}
